frontend/bot/src/Coda.re

```diff
@@ -24,3 +24,48 @@ let sendPayment = (~from, ~to_, ~amount, ~fee) => {
     (),
   );
 };
+
+module Unlock = [%graphql
+  {| mutation unlock($password: String, $publicKey: PublicKey) {
+      unlockWallet(input: {password: $password, publicKey: $publicKey}) {
+          publicKey
+        }
+  } |}
+];
+
+let unlock = (~publicKey, ~password) => {
+  let log = (w, fmt) => Logger.log("Unlock", w, fmt);
+
+  ReasonUrql.Client.executeMutation(
+    ~client=Graphql.client,
+    ~request=
+      Unlock.make(
+        ~password,
+        ~publicKey=Graphql.Encoders.publicKey(publicKey),
+        (),
+      ),
+    (),
+  )
+  |> Wonka.forEach((. {ReasonUrql.Client.Types.response}) =>
+       switch (response) {
+       | Data(_) => log(`Info, "Unlock successful of %s", publicKey)
+       | Error(e) =>
+         log(
+           `Error,
+           "Unlock failed for %s, error: %s",
+           publicKey,
+           Js.String.make(e),
+         )
+       | NotFound => log(`Error, "Got 'NotFound' unlocking %s", publicKey)
+       }
+     );
+};
+
+let unlockOpt = (~publicKey, ~password) => {
+  switch (publicKey, password) {
+  | (Some(publicKey), Some(password)) => unlock(~publicKey, ~password)
+  | (Some(publicKey), None) =>
+    Logger.log("Unlock", `Error, "No password provided for %s", publicKey)
+  | (None, _) => () // Nothing to unlock
```

Comment:
should we log saying please provide a public key to unlock?

Reply:
No, if you don't provide a key, the service doesn't run, by design, as a way of turning things off.